### PR TITLE
prevent the Logger from writing sensitive values

### PR DESCRIPTION
**Pull request by blizzz**
* [x] requires https://github.com/nextcloud/server/pull/33398
* [ ] ~~ensure backwards compat if necessary~~ only backports

### CODE REVIEW

**Review by nickvergessen — APPROVED** [review]

File: lib/AppInfo/Application.php [hunk 1]
```diff
@@ -30,13 +30,17 @@
 use OCP\AppFramework\Bootstrap\IBootContext;
 use OCP\AppFramework\Bootstrap\IBootstrap;
 use OCP\AppFramework\Bootstrap\IRegistrationContext;
+use Office365\Runtime\Auth\AuthenticationContext;
+use Office365\Runtime\Auth\SamlTokenProvider;
```
Comment (nickvergessen) [inline lib/AppInfo/Application.php:34]: Seeing both paths being in the same namespace, does it make sense to allow giving class names only or even namespaces only?

Comment (blizzz) [reply lib/AppInfo/Application.php:34]: Both might be too broad in other cases.

Comment (nickvergessen) [reply lib/AppInfo/Application.php:34]: Well that's why I meant "Allow", but yeah then let's keep it like this.
I just fear the moment they change an internal method name or something and it logs again

Comment (blizzz) [reply lib/AppInfo/Application.php:34]: > Well that's why I meant "Allow", but yeah then let's keep it like this. I just fear the moment they change an internal method name or something and it logs again

Absolutely. But that can happen to namespaces as well as class names (also happened in this lib in the past).

The only sustainable solution will arrive with PHP 8.2's sensitive parameter support.

Comment (blizzz) [reply lib/AppInfo/Application.php:34]: @nickvergessen added an integrattion test that should act as safeguard nevertheless

**Comment by blizzz** [comment]
For the record: integration tests succeeds locally against the `enh/noid/sensitive-methods-apps` branch of the  server.

**Comment by PVince81** [comment]
server PR merged, please rebase/adjust

**Comment by blizzz** [comment]
> server PR merged, please rebase/adjust

rerun failing tests against current master. All green now → merging

**Pull request merged**